llvm/lib/Target/AArch64/AArch64SVEInstrInfo.td

```diff
@@ -3750,6 +3750,23 @@ let Predicates = [HasSVE2orSME] in {
 
   // SVE2 extract vector (immediate offset, constructive)
   def EXT_ZZI_B : sve2_int_perm_extract_i_cons<"ext">;
+
+  // zext(cmpeq(bsl(x, y, z), splat(0))) -> nbsl(x, y, z)
```

Comment:
The instruction looks like it should be `not(or(and(x, z), and(y, not(z)))`, not the same as a "`cnot`" instruction. I think you can use vnot in the tablegen pattern, and the ir will be `xor "bsl", -1`.

Reply:
It may be possible to pass a pattern fragment to sve2_int_bitwise_ternary_op, like bsl already does.

Reply:
Done.